scripts/dependency_updater.py

```diff
@@ -0,0 +1,536 @@
+                                    edges {{
+                                        node {{
+                                            oid
+                                            message
+                                            url
+                                        }}
+                                    }}
+                                }}
+                            }}
+                        }}
+                    }}
+                }}
+            }}
+        """)
+
+    query = f"query {{ {''.join(query_parts)} }}"
+    headers = {
+        'Authorization': f'Bearer {gh_token}',
+        'Content-Type': 'application/json'
+    }
+
+    try:
+        response = session.post(github_api_url, json={'query': query}, headers=headers)
+        response.raise_for_status()
+        json_data = response.json()
+        data = json_data.get('data')
+        if data is not None and bool(data):  # Ensure 'data' is not None and not empty
+            logging.debug(f'GraphQL data response:\n{json.dumps(data, indent=2)}')
+            return data
+        else:
+            logging.error(f'GraphQL query returned errors: {json_data}')
+            return None
+    except Exception as e:
+        logging.error(f'Error fetching repository metadata: {e}')
+        return None
+
+def calculate_checksum(cachefile, sha_regex):
+    if sha_regex:
+        logging.debug(f'Searching with regex {sha_regex} in file {cachefile}')
+        with open(f'cache/{cachefile}', 'r') as f:
+            for line in f:
+                if sha_regex == 'simple': # Only sha is present in the file
+                    pattern = re.compile(SHA256REGEX)
+                else:
+                    pattern = re.compile(rf'(?:{SHA256REGEX}.*{sha_regex}|{sha_regex}.*{SHA256REGEX})') # Sha may be at start or end
+                match = pattern.search(line)
+                if match:
+                    checksum = match.group(1) or match.group(2)
+                    logging.debug(f'Matched line: {line.strip()}')
+                    return checksum
+    else: # binary
+        sha256_hash = hashlib.sha256()
+        with open(f'cache/{cachefile}', 'rb') as f:
+            for byte_block in iter(lambda: f.read(4096), b''):
+                sha256_hash.update(byte_block)
+        checksum = sha256_hash.hexdigest()
+        return checksum
+
+def download_file_and_get_checksum(component, arch, url_download, version, sha_regex, session):
+    logging.info(f'Download URL {url_download}')
+    cache_file = f'{component}-{arch}-{version}'
+    if os.path.exists(f'cache/{cache_file}'):
+        logging.info(f'Using cached file for {url_download}')
+        return calculate_checksum(cache_file, sha_regex)
+    try:
+        response = session.get(url_download, timeout=10)
+        response.raise_for_status()
+        with open(f'cache/{cache_file}', 'wb') as f:
+            f.write(response.content)
+        logging.info(f'Downloaded and cached file for {url_download}')
```

Comment:
I don't think we want to do caching (at least at start) It adds some footguns + I'd rather have a more predictable output from multiples "runners" (CI, locally, etc).